thirdeye-persistence/src/main/java/ai/startree/thirdeye/datalayer/database/DatabaseTaskService.java

```diff
@@ -0,0 +1,82 @@
+package ai.startree.thirdeye.datalayer.database;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.MetricRegistry;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import java.sql.Connection;
+import java.sql.SQLException;
+import javax.sql.DataSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Singleton
+public class DatabaseTaskService {
```

Comment:
Don't quite understand this. What is the main layer of separation here?

Reply:
I was thinking the `DatabaseService` would decide which db operation to perform and let `DatabaseTaskService` worry about the transactional aspect of executing the operations and handling the connection etc.

Reply:
Nice thought here. However, let's tackle this in a separate PR